prov/tcp/src/tcpx_rma.c

```diff
@@ -106,12 +106,14 @@ static ssize_t tcpx_rma_readmsg(struct fid_ep *ep, const struct fi_msg_rma *msg,
 	assert(msg->rma_iov_count <= TCPX_IOV_LIMIT);
 
 	send_entry = tcpx_xfer_entry_alloc(tcpx_cq, TCPX_OP_READ_REQ);
-	if (!send_entry)
+	if (!send_entry) {
+		tcpx_xfer_entry_release(tcpx_cq, send_entry);
 		return -FI_EAGAIN;
+	}
 
 	recv_entry = tcpx_xfer_entry_alloc(tcpx_cq, TCPX_OP_READ_RSP);
 	if (!recv_entry) {
-		tcpx_xfer_entry_release(tcpx_cq, send_entry);
+		tcpx_xfer_entry_release(tcpx_cq, recv_entry);
```

Comment:
Same here, we failed to get a recv_entry, so we're freeing the previously allocated send_entry,